wdl/transforms/draft3/src/main/scala/wdl/draft3/transforms/wdlom2wdl/WdlWriterImpl.scala

```diff
@@ -0,0 +1,438 @@
+    override def toWdlV1(a: OutputsSectionElement): String = {
+      s"""output {
+         |${indentAndCombine(a.outputs.map(_.toWdlV1))}}""".stripMargin
+    }
+  }
+
+  implicit val outputDeclarationElementWriter: WdlWriter[OutputDeclarationElement] = new WdlWriter[OutputDeclarationElement] {
+    override def toWdlV1(a: OutputDeclarationElement): String = {
+      s"${a.typeElement.toWdlV1} ${a.name} = ${a.expression.toWdlV1}"
+    }
+  }
+
+  implicit val functionCallElementWriter: WdlWriter[FunctionCallElement] = new WdlWriter[FunctionCallElement] {
+    override def toWdlV1(a: FunctionCallElement): String = a match {
+      case _: StdoutElement.type => "stdout()"
+      case _: StderrElement.type => "stderr()"
+      case a: OneParamFunctionCallElement => a.toWdlV1
+      case a: OneOrTwoParamFunctionCallElement => a.toWdlV1
+      case a: TwoParamFunctionCallElement => a.toWdlV1
+      case a: Sub => s"sub(${a.input.toWdlV1}, ${a.pattern.toWdlV1}, ${a.replace.toWdlV1})"
+    }
+  }
+
+  implicit val oneParamFunctionCallElementWriter: WdlWriter[OneParamFunctionCallElement] = new WdlWriter[OneParamFunctionCallElement] {
+    override def toWdlV1(a: OneParamFunctionCallElement): String = {
+      def functionCall(name: String) = s"$name(${a.param.toWdlV1})"
+
+      a match {
+        case _: ReadLines =>    functionCall("read_lines")
+        case _: ReadTsv =>      functionCall("read_tsv")
+        case _: ReadMap =>      functionCall("read_map")
+        case _: ReadObject =>   functionCall("read_object")
+        case _: ReadObjects =>  functionCall("read_objects")
+        case _: ReadJson =>     functionCall("read_json")
+        case _: ReadInt =>      functionCall("read_int")
+        case _: ReadString =>   functionCall("read_string")
+        case _: ReadFloat =>    functionCall("read_float")
+        case _: ReadBoolean =>  functionCall("read_boolean")
+        case _: WriteLines =>   functionCall("write_lines")
+        case _: WriteTsv =>     functionCall("write_tsv")
+        case _: WriteMap =>     functionCall("write_map")
+        case _: WriteObject =>  functionCall("write_object")
+        case _: WriteObjects => functionCall("write_objects")
+        case _: WriteJson =>    functionCall("write_json")
+        case _: Range =>        functionCall("range")
+        case _: Transpose =>    functionCall("transpose")
+        case _: Length =>       functionCall("length")
+        case _: Flatten =>      functionCall("flatten")
+        case _: SelectFirst =>  functionCall("select_first")
+        case _: SelectAll =>    functionCall("select_all")
+        case _: Defined =>      functionCall("defined")
+        case _: Floor =>        functionCall("floor")
+        case _: Ceil =>         functionCall("ceil")
+        case _: Round =>        functionCall("round")
+        case _: Glob =>         functionCall("glob")
```

Comment:
similar DRYing comment as above, minus the sandwich

Reply:
The big block of same-named functions did feel wrong to me aesthetically. Yours is the obvious solution, which apparently eluded me on two separate occasions